runtime/server/queries_metrics.go

```diff
@@ -484,6 +490,114 @@ func (s *Server) MetricsViewTimeRanges(ctx context.Context, req *runtimev1.Metri
 	}, nil
 }
 
+func (s *Server) MetricsViewAnnotations(ctx context.Context, req *runtimev1.MetricsViewAnnotationsRequest) (*runtimev1.MetricsViewAnnotationsResponse, error) {
+	observability.AddRequestAttributes(ctx,
+		attribute.String("args.instance_id", req.InstanceId),
+		attribute.String("args.metric_view", req.MetricsViewName),
+		attribute.String("args.measures", strings.Join(req.Measures, ",")),
+	)
+
+	if !auth.GetClaims(ctx).CanInstance(req.InstanceId, auth.ReadMetrics) {
+		return nil, ErrForbidden
+	}
+	s.addInstanceRequestAttributes(ctx, req.InstanceId)
+
+	var limit *int64
+	if req.Limit != 0 {
+		limit = &req.Limit
+	}
+
+	var offset *int64
+	if req.Offset != 0 {
+		limit = &req.Offset
+	}
+
+	tr := metricsview.TimeRange{}
+	if req.TimeRange != nil {
+		tr = metricsview.TimeRange{
+			Start:         req.TimeRange.Start.AsTime(),
+			End:           req.TimeRange.End.AsTime(),
+			Expression:    req.TimeRange.Expression,
+			IsoDuration:   req.TimeRange.IsoDuration,
+			IsoOffset:     req.TimeRange.IsoOffset,
+			RoundToGrain:  metricsview.TimeGrainFromProto(req.TimeRange.RoundToGrain),
+			TimeDimension: req.TimeRange.TimeDimension,
+		}
+	}
+
+	res, err := s.runtime.Resolve(ctx, &runtime.ResolveOptions{
+		InstanceID: req.InstanceId,
+		Resolver:   "annotations",
+		ResolverProperties: map[string]any{
+			"metrics_view": req.MetricsViewName,
+			"measures":     req.Measures,
+			"priority":     req.Priority,
+			"time_range":   tr,
+			"time_grain":   req.TimeGrain,
+			"limit":        limit,
+			"offset":       offset,
+		},
```

Comment:
Consider using `metricsview.AnnotationsQuery{}` and `.AsMap()` for type safety (in case we rename properties one place and forget elsewhere)